app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailViewModel.kt

```diff
@@ -0,0 +1,139 @@
+import com.wire.android.navigation.NavigationManager
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
+import com.wire.kalium.logic.feature.user.UpdateEmailUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ChangeEmailViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+    private val updateEmail: UpdateEmailUseCase,
+    private val getSelf: GetSelfUserUseCase,
+) : ViewModel() {
+
+    var state: ChangeEmailState by mutableStateOf(
+        ChangeEmailState(
+            isEmailTextEditEnabled = true
+        )
+    )
+        @VisibleForTesting
+        set
+
+    private var currentEmail: String? = null
+
+    init {
+        viewModelScope.launch {
+            getSelf().firstOrNull()?.email?.let {
+                currentEmail = it
+                state = state.copy(email = TextFieldValue(it))
+            } ?: onBackPressed()
+        }
+    }
+
+    fun onSaveClicked() {
+        state = state.copy(saveEnabled = false, isEmailTextEditEnabled = false)
+        viewModelScope.launch {
+            when (updateEmail(state.email.text)) {
+                UpdateEmailUseCase.Result.Failure.EmailAlreadyInUse -> state =
+                    state.copy(
+                        isEmailTextEditEnabled = true,
+                        saveEnabled = false,
+                        error = ChangeEmailState.EmailError.TextFieldError.AlreadyInUse
+                    )
+
+                UpdateEmailUseCase.Result.Failure.InvalidEmail -> state =
+                    state.copy(
+                        isEmailTextEditEnabled = true,
+                        saveEnabled = false,
+                        error = ChangeEmailState.EmailError.TextFieldError.InvalidEmail
+                    )
+
+                is UpdateEmailUseCase.Result.Failure.GenericFailure -> state =
+                    state.copy(
+                        isEmailTextEditEnabled = true,
+                        saveEnabled = false,
+                        error = ChangeEmailState.EmailError.TextFieldError.InvalidEmail
+                    )
+
+                is UpdateEmailUseCase.Result.Success.VerificationEmailSent -> onUpdateEmailSuccess()
+                is UpdateEmailUseCase.Result.Success.NoChange -> onBackPressed()
+            }
+        }
+    }
+
+    private fun onUpdateEmailSuccess() {
+        viewModelScope.launch {
+            navigationManager.navigate(
+                NavigationCommand(
+                    NavigationItem.VerifyEmailAddress.getRouteWithArgs(listOf(state.email.text)),
+                    BackStackMode.REMOVE_CURRENT
+                )
+            )
+        }
+    }
+
+    fun onEmailChange(newEmail: TextFieldValue) {
+        val cleanEmail = newEmail.text.trim()
+        val isValidEmail = cleanEmail.contains('@')
```

Comment:
Question, is it possible to use something like to validate https://developer.android.com/reference/android/util/Patterns#EMAIL_ADDRESS

Just asking without context, maybe the `@` Symbol is the requirement needed at Wire.